_posts/2015-12-16-API-Blueprint-Future.md

```diff
@@ -0,0 +1,100 @@
+
+I am still shocked how much these core values resonated with the community. To ask programmers to write in plain text instead of JSON or XML was the bet we took. And the bet wasn't only about writing in freeform text. We also asked to not generate the API description from code. Instead we suggested it is better to take the effort and scope out an API definition before coding.
+
+After three years I can say our bet was well worth taking. API Blueprint is now growing faster then ever. It has way over 3,000 star gazers on [GitHub][1] and almost 200,000 blueprints in Apiary. It's now used everywhere from IoT devices to banks.
+
+### The most important feature
+
+So where is API Blueprint going next? Before looking into the future let's start with the present.
+
+In 2015, we have spent most of the year building only one feature—the description of body parameters. This feature is also know as the [MSON syntax][2].
+
+Only one feature? Why?
+
+Because this feature brings the single most important concept in understanding an API. It introduces the description of the application data model domain and its semantics.
+
+When encountering a new system, I first need to understand its semantics. What is the data in it, and what I can do with it? Only then I can start asking how to interact with the system. This may sound trivial, but it isn't. If my customers do not know what my API is offering, why should they bother using it?
+
+I am already hinting here where the API Blueprint is heading.
+
+At the pinnacle of every API design is the semantics of its application domain. Once I understand **what** an API has to offer I can proceed forward to **how** to make an use of it.
+
+### Abstraction
+
+While I am pondering about the **what** of an API, I don't want to worry about JSONs, URLs, protocols or other details. These decisions can wait after I define what the API is actually about.
+
+Abstraction plays a key role in any good software design. It makes design of complex systems possible. It encapsulates behavior, decouples elements and enables modularity.
+
+Let's look at the levels of abstraction when designing an API and its implementation.
+
+![][3]
+
+With MSON, API Blueprint took a major step towards modeling application data.
+
+But make no mistake—MSON isn't another syntax for JSON. MSON is agnostic to serialization media types. With MSON, I can describe the data and defer the decision whether they will be send as JSON, XML or [HAL][4] over the wire.
+
+### Plans
+
+Here are the upcoming steps for API Blueprint.
+
+To finish the data description, both URI parameters and HTTP headers are going to use the MSON syntax. The same syntax of parameters, headers and attributes allows the reuse of data models. It will also enable more powerful tools thanks to the data connection. [[link][5]]
```

Comment:
@zdne If I may offer a SEO and search and reading-tools (accessibility) improvement over using "link" (as text for hyperlink) - this is the worst thing you can do. Having a link inside a text is much much more relevant and also such link carries context of that sentence.
I'd suggest changing this paragraph into:

```markdown
To finish the data description, both [URI parameters and HTTP headers are going to use the MSON syntax][link[5]]. The same syntax of parameters, headers and attributes allows the reuse of data models. It will also enable more powerful tools thanks to the data connection.
```

Similar approach also applies for all other paragraphs in the `Plans` section.